third-party/rsutils/include/rsutils/deferred.h

```diff
@@ -42,10 +46,10 @@ class deferred
     }
 
     // Prevent the deferred call from ever being invoked
-    void cancel() { _deferred = nullptr; }
+    void reset() { _deferred = {}; }
 
     // Returns the deferred function so you can move it somewhere else
-    fn && detach() { return std::move( _deferred ); }
+    fn detach() { return std::move( _deferred ); }
```

Comment:
I am not sure about the state of `_deferred` after the move. Acording to https://stackoverflow.com/questions/13680587/move-semantic-with-stdfunction it will be in an unspecified state, that might not be empty.
If subscription is `detach`ed and then `cancel`ed it might call the detached funciton?

Reply:
It says "Unless otherwise specified" -- meaning if a move constructor or operator was not defined. In the case of std::function, it is defined. It is true that, up to C++20, it is not guaranteed to be valid. But this is tested in the unit-test. And between us, we know it resets to nothing...
```
void _Reset_move(_Func_class&& _Right) noexcept { // move _Right's stored object
        if (!_Right._Empty()) {
            if (_Right._Local()) { // move and tidy
                _Set(_Right._Getimpl()->_Move(&_Mystorage));
                _Right._Tidy();
            } else { // steal from _Right
                _Set(_Right._Getimpl());
                _Right._Set(nullptr);
            }
        }
    }
function(function&& _Right) noexcept {
        this->_Reset_move(_STD move(_Right));
    }
```
Anyway, if this is a problem then it's a problem in `deferred` which was there before too. Let's talk?

Reply:
> If subscription is `detach`ed and then `cancel`ed it might call the detached funciton?

A detach will do `_d.detach()`, meaning `_d` should now be nothing. If you then `cancel()` then `_d` is nothing and nothing should happen. Right?